api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -92,7 +96,7 @@
  * @since 4.0.0
  */
 @ApiStatus.NonExtendable
-public interface Component extends ComponentBuilderApplicable, ComponentLike, Examinable, HoverEventSource<Component> {
+public interface Component extends ComponentBuilderApplicable, ComponentLike, Examinable, HoverEventSource<Component>, Iterable<Component> {
```

Comment:
imo `Component` should not directly implement `Iterable` (at least, not from the API side) -- it's a tree, not a sequence. Having separate methods that request a view as an `Iterable` or `Iterator` makes it clear that the flat `Iterator` is one of many possible views for a component.

Reply:
Yup that makes sense actually. I'd almost err towards removing this alongside the methods that don't take an iteration type to force people to decide what sort of iteration they want to do.

Reply:
I agree with @zml2008 - implementing `Iterable` seems inappropriate for Components